File: Automation/FC/CommonTestSteps/EmailServiceTests.cs
using SeleniumCore;
using Utilities;

namespace CommonTestSteps
{
    public class EmailServiceTests
    {
        public static void ObtainRandomEmailAddress()
        {
            TestInMemoryParameters.Instance.ElementTimeout = "45";
            TestInMemoryParameters.Instance.PageLoadTimeout = "180";
            SharedSteps.Containers.EmailService.CheckAvailability(TestInMemoryParametersShared.Instance.EmailServices);
            SharedSteps.Containers.EmailService.Initialize();
            SharedSteps.Containers.Global.GoToMailService();
            SharedSteps.Containers.EmailService.GetEmailAddress();
            SharedSteps.Containers.Global.CloseBrowser();
        }

        public static void GetVerificationCode()
        {
            TestInMemoryParametersShared.Instance.VerificationSenderAddress = "msonlineservicesteam@microsoftonline.com";
            TestInMemoryParametersShared.Instance.VerificationCodeText = "Your code is:";
            SharedSteps.Containers.Global.GoToMailServiceByName();
            SharedSteps.Containers.EmailService.GetVerificationCodeFromEmail();
            SharedSteps.Containers.Global.CloseBrowser();
        }
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/test/java/testSuites/shared/tests/EmailServiceTests.java
package testSuites.shared.tests;

import common.core.TestInMemoryParameters;
import testSuites.shared.steps.CommonBrowserActions;
import testSuites.shared.steps.emailServices.EmailServiceTestSteps;

public class EmailServiceTests {
    public static void obtainRandomEmailAddress() {
        EmailServiceTestSteps emailServiceTestSteps = new EmailServiceTestSteps();
        emailServiceTestSteps.checkAvailability(TestInMemoryParameters.getInstance().emailServices);
        emailServiceTestSteps.initialize();
        CommonBrowserActions.goToDefaultEmailService();
        emailServiceTestSteps.getEmailAddress();
        CommonBrowserActions.closeBrowser();
    }
}
